src/server.ts

```diff
@@ -749,3 +752,69 @@ function findCreatedAccountSequenceInTransactionMeta(
 
   throw new Error("No account created in transaction");
 }
+
+
+// TODO - remove once rpc updated to
+// append expiration entry per data LK requested onto server-side response 
+// https://github.com/stellar/soroban-tools/issues/1010
+function mergeResponseExpirationLedgers(ledgerEntriesResponse: SorobanRpc.RawGetLedgerEntriesResponse,
+  requestedKeys: xdr.LedgerKey[]
+): SorobanRpc.RawGetLedgerEntriesResponse {
+  const expirationKeyToRawEntryResult = new Map<String, SorobanRpc.RawLedgerEntryResult>();
+  const requestedKeyXdrs = new Set<String>(requestedKeys.map(requestedKey =>
+    requestedKey.toXDR('base64')));
+
+  (ledgerEntriesResponse.entries ?? []).forEach(rawEntryResult => {
+    if (!rawEntryResult.key) {
+      // don't interpret raw ledger entry data here, just pass it through to the outgoing response as-is
+      expirationKeyToRawEntryResult.set(Math.random().toString(), rawEntryResult)
+      return;
+    }
+    const parsedKey = xdr.LedgerKey.fromXDR(rawEntryResult.key, 'base64');
+    
+    if (parsedKey.switch().value !== xdr.LedgerEntryType.expiration().value ||
+        requestedKeyXdrs.has(rawEntryResult.key)) {
+      let keyHash = hash(parsedKey.toXDR()).toString()
+      let rawEntryResultLookup = expirationKeyToRawEntryResult.get(keyHash);
+
+      if (!rawEntryResultLookup) {
+        rawEntryResultLookup = rawEntryResult;
+        expirationKeyToRawEntryResult.set(keyHash, rawEntryResultLookup);
+      } 
+      rawEntryResultLookup.key = rawEntryResult.key;
+      rawEntryResultLookup.xdr = rawEntryResult.xdr;
+      rawEntryResultLookup.lastModifiedLedgerSeq = rawEntryResult.lastModifiedLedgerSeq;
+      return;
+    } 
+
+    let keyHash = parsedKey.expiration().keyHash().toString();
+    let rawEntryResultLookup = expirationKeyToRawEntryResult.get(keyHash);
+
+    if (!rawEntryResultLookup) {
+      rawEntryResultLookup = {
+        key: "", 
+        xdr: ""
+      };
+      expirationKeyToRawEntryResult.set(keyHash, rawEntryResultLookup);
+    }
+    rawEntryResultLookup.expiration = xdr.LedgerEntryData
+      .fromXDR(rawEntryResult.xdr, 'base64')
+      .expiration()
+      .toXDR('base64');
+  });
+  
+  ledgerEntriesResponse.entries = [...expirationKeyToRawEntryResult.values()];
+  return ledgerEntriesResponse;
+}
+
+// TODO - remove once rpc updated to
+// include expiration entry on responses for any data LK's requested  
+// https://github.com/stellar/soroban-tools/issues/1010
+function expandRequestIncludeExpirationLedgers(
+  keys: Array<xdr.LedgerKey>
+): Array<xdr.LedgerKey> {
```

Comment:
you can use `[]` here instead of `Array` - that's what the codebase does elsewhere afaict

```suggestion
function expandRequestIncludeExpirationLedgers(keys: xdr.LedgerKey[]): xdr.LedgerKey[] {
```